python/src/main/python/drivers/piglit-worker.py

```diff
@@ -248,34 +254,43 @@ def run_image_job(frag_file: str, json_file: str, status_file: str,
     shader_test_file = graphicsfuzz_piglit_converter.get_shader_test_from_job(json_file)
 
     try:
-        runspv.log('Creating shader_test file...')
+        gfuzz_common.log('Creating shader_test file...')
         graphicsfuzz_piglit_converter.main_helper(arglist)
     except Exception as ex:
-        runspv.log('Could not create shader_test from the given job.')
+        gfuzz_common.log('Could not create shader_test from the given job.')
         raise ex
 
-    status = STATUS_SUCCESS
+    shader_runner_cmd_list = catchsegv_cmd() + shader_runner_cmd() + \
+        [shader_test_file, SHADER_RUNNER_ARG_AUTO,
+         SHADER_RUNNER_ARG_UNIFORMS, SHADER_RUNNER_ARG_SUBTESTS]
 
-    shader_runner_cmd_list = shader_runner_cmd() + [shader_test_file, SHADER_RUNNER_ARG_AUTO,
-                                                    SHADER_RUNNER_ARG_UNIFORMS,
-                                                    SHADER_RUNNER_ARG_SUBTESTS]
     if not skip_render:
         shader_runner_cmd_list.append(SHADER_RUNNER_ARG_PNG)
-    try:
-        runspv.subprocess_helper(shader_runner_cmd_list, timeout=TIMEOUT, verbose=True)
-    except subprocess.TimeoutExpired:
-        status = STATUS_TIMEOUT
-    except subprocess.CalledProcessError:
-        status = STATUS_CRASH
 
-    runspv.log('STATUS: ' + status)
+    status = gfuzz_common.run_catchsegv(shader_runner_cmd_list, timeout=TIMEOUT, verbose=True)
 
     # Piglit throws the output PNG render into whatever the current working directory is
     # (and there's no way to specify a location to write to) - we need to move it to wherever our
     # output is.
 
-    if os.path.isfile(PNG_FILENAME):
-        shutil.move(PNG_FILENAME, png_file)
+    if not skip_render and status == STATUS_SUCCESS:
+        # An image was rendered, so we need to check for nondet. We do this by renaming the
+        # rendered image, rendering a second image, and using filecmp to compare the files.
+        assert os.path.isfile(PNG_FILENAME)
+        os.rename(PNG_FILENAME, COMPARE_PNG_FILENAME)
+        status = gfuzz_common.run_catchsegv(shader_runner_cmd_list, timeout=TIMEOUT, verbose=True)
+        # Something is horribly wrong if shader crashes/timeouts are inconsistent per shader.
+        if not status == STATUS_SUCCESS:
+            raise AssertionError("Shader inconsistently fails - check your graphics drivers?")
```

Comment:
Raising any exception in `run_image_job` makes the worker return an UNEXPECTED_ERROR since the status file doesn't exist - I figured AssertionError would be the most descriptive exception type to use here. Is there a different exception type I should use?

Reply:
That sounds good. I might be misreading this (on GitHub), but it looks like this is not the case; if the status file missing, nothing happens and the result is not really set.

Reply:
Never mind! I see it now.